packages/inferno-animation/src/utils.ts

```diff
@@ -0,0 +1,206 @@
+declare global {
+  // Setting `window.__DEBUG_ANIMATIONS__ = true;` disables animation timeouts
+  // allowing breakpoints in animations for debugging.
+  // https://mariusschulz.com/blog/declaring-global-variables-in-typescript
+  var __DEBUG_ANIMATIONS__: Boolean;
+}
+
+export function addClassName (node: HTMLElement, className: string) {
+  if (className) {
+    const tmp = className.split(' ')
+    for (let i=0; i < tmp.length; i++) {
+      node.classList.add(tmp[i])
+    }
+  }
+}
+
+export function removeClassName (node: HTMLElement, className: string) {
+  if (className) {
+    const tmp = className.split(' ')
+    for (let i=0; i < tmp.length; i++) {
+      node.classList.remove(tmp[i])
+    }
+  }
+}
+
+export function forceReflow () {
+  return document.body.clientHeight;
+}
+
+export function setDisplay(node: HTMLElement, value?: string) {
+  var oldVal = node.style.getPropertyValue('display')
+  if (oldVal !== value) {
+    if (value !== undefined) {
+      node.style.setProperty('display', value)
+    }
+    else {
+      node.style.removeProperty('display')
+      _cleanStyle(node);
+    }
+  }
+  return oldVal
+}
+
+function _cleanStyle(node: HTMLElement) {
+  if (!node.style) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute
+    node.removeAttribute('style')
+  }
+}
+
+export function getDimensions(node: HTMLElement) {
+  var tmpDisplay = node.style.getPropertyValue('display')
+  var isDisplayNone = window.getComputedStyle(node).getPropertyValue('display') === 'none'
+  if (isDisplayNone) {
+    node.style.setProperty('display', 'block')
```

Comment:
doesn't this break flex box styling?

Reply:
Yes, will fix (code created when flex wasn't widely supported)

Reply:
Okay, this is tricky. I am not sure we can know if it is meant to be flex or block here. This was added to support bootstrap 4 style animations in inferno-bootstrap. https://github.com/jhsware/inferno-bootstrap/blob/master/src/Modal/AnimateModal.js
I would like to keep it but add a comment about why it exists so it potentially can be removed in the future.

Reply:
WON'T FIX comment in 587866ca3